### DISCUSSION TITLE
Json Codec with Fields out of Order

### DISCUSSION THREAD

**Discussion by arsentiikaptsan** [activej/activej#171] [Q&A]
It seems like JsonUtils.fromJson is sensitive to order of fields in json. Also it does not handle missing or unknown fields gracefully. Can I fix this with proper configuration?

**Comment by eduard-vasinskyi** [comment]
Hello, @arsentiikaptsan

[ActiveJ Codec](https://github.com/activej/activej/tree/master/util-codec) module is designed to encode/decode data with a predefined structure. The structure has to be ordered and may not contain unknown fields.
We use codecs in some of our components, however, ActiveJ is an agnostic framework in terms of JSON processing. You may use ActiveJ in conjunction with any existing JSON library that suits your needs ([Dsl-Json](https://github.com/ngs-doo/dsl-json), [Jackson](https://github.com/FasterXML/jackson), etc).

**Comment by arsentiikaptsan** [comment]
Ok, understood, thank you!
